### PR TITLE
LL-837 Initial version of the updated refactor of modals

### PR DESCRIPTION
**Pull request by juan-cortes**
> Updated version of https://github.com/LedgerHQ/ledger-live-desktop/pull/1358

It's an optimistically complete version of the linked pr from last year, although I'm going to need some more eyes to locate bugs/regressions/weird stuff. This pr opens the door to different animations, centered modals (done), and much more.

@gre @meriadec No idea why I can't save the account settings, I'd appreciate a look from either.

- [x] Password Modal
- [x] ResetFallbackModal
- [x] ResetLedgerLive
- [x] Onboarding = Genuine check
- [x] Onboarding = Technical data
- [x] Onboarding = Analytics
- [x] SendFlow
- [x] ReceiveFlow
- [x] ~ Confirm Modal
    - Locked screen, confirm delete
    - Clean button Modal
    - Error Modal
- [x] AppList (install & uninstall flow)
- [x] PasswordLock = Disable
- [x] PasswordLock = Enable
- [x] Add Account
- [x] Release notes
- [x] Export
- [x] AccountSettings - Can't save?
- [ ] Testing this mofo

### CODE REVIEW

File: src/components/SettingsPage/PasswordModal.js [hunk 1]
```diff
@@ -36,6 +37,13 @@ const INITIAL_STATE = {
 class PasswordModal extends PureComponent<Props, State> {
   state = INITIAL_STATE
 
+  componentWillReceiveProps(nextProps: Props) {
+    if (!nextProps.isOpened) {
+      // CLean the state?
+      this.setState(prevState => ({ ...prevState, ...INITIAL_STATE }))
```
Comment (meriadec) [inline src/components/SettingsPage/PasswordModal.js]: `this.setState(INITIAL_STATE)` should be enough

**Comment by juan-cortes** [comment]
@Arnaud97234 Scrolling issue on operation details and ghost footer on modals without content for footer should be addressed with latest commit, thanks for finding them.
image

**Comment by Arnaud97234** [comment]
**Bug found**
- On Fedora and Windows, there is a visual glitch at the bottom of Modals having a scrollbar (For instance Release note).

releasenote_fedorawindows


- Edit Account Modal (Account -> Account settings)
Regarding to application window size, when unfolding "Advanced logs" a text area is displayed.
XRP account : Text area covers the modal's bottom (Windows).
Modal's button are outside of the modal (Fedora and Macos).

xrp_editaccount_fedoramacos
xrp_accountssettings_windows
xrp_accountsettings_macos
tbtc_accountsettings_windowsmacos

- Settings -> Password lock modal
Miss autofocus on save button

- Trade safely modal, Onboarding -> Analytics modal, Onboarding -> Technical data
Bold text between title and bullet points is centered (before it was stuck on the left of application window).
The close button (X) is not present.

**Comment by gre** [comment]
need rebase

**Comment by juan-cortes** [comment]
I'll rebase tonight, don't know if I can do anything about the bugs on fedora but I'll look into it

**Comment by meriadec** [comment]
I can test / debug on monday with @Arnaud97234

**Comment by juan-cortes** [comment]
@gre rebased

**Comment by Arnaud97234** [comment]
Edit account modals are OK now, thanks @meriadec .

Bugs remaining :
- Settings -> Password lock modal
Miss autofocus on save button

- Trade safely modal, Onboarding -> Analytics modal, Onboarding -> Technical data
Bold text between title and bullet points is centered (before it was stuck on the left of application window).
The close button (X) is not present.

- All Modals :
Padding top is smaller that the one currently present on production app
addaccountsmodal_prod
addaccountsmodal_refactorbranch

.

**Comment by juan-cortes** [comment]
@Arnaud97234 
- Pressing enter on the password confirmation now triggers the same action as pressing the button
- Share Analytics modal now has the close button
- Technical Data modal now has the close button
- Trade safely / Disclaimer modal now has the close button

Let me know if you find any other thing out of place, it should also be a quick fix.

missing modal close buttons

**Comment by juan-cortes** [comment]
🤞

**Pull request merged**